static/app/views/insights/agentMonitoring/views/onboarding.tsx

```diff
@@ -0,0 +1,505 @@
+import {
+  type Configuration,
+  type StepProps,
+  StepTitles,
+  StepType,
+  TabbedCodeSnippet,
+} from 'sentry/components/onboarding/gettingStartedDoc/step';
+import {
+  DocsPageLocation,
+  type DocsParams,
+  ProductSolution,
+} from 'sentry/components/onboarding/gettingStartedDoc/types';
+import {useSourcePackageRegistries} from 'sentry/components/onboarding/gettingStartedDoc/useSourcePackageRegistries';
+import {useLoadGettingStarted} from 'sentry/components/onboarding/gettingStartedDoc/utils/useLoadGettingStarted';
+import Panel from 'sentry/components/panels/panel';
+import PanelBody from 'sentry/components/panels/panelBody';
+import {agentMonitoringPlatforms} from 'sentry/data/platformCategories';
+import platforms, {otherPlatform} from 'sentry/data/platforms';
+import {t, tct} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import pulsingIndicatorStyles from 'sentry/styles/pulsingIndicator';
+import {space} from 'sentry/styles/space';
+import type {PlatformKey, Project} from 'sentry/types/project';
+import {getSelectedProjectList} from 'sentry/utils/project/useSelectedProjectsHaveField';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import useProjects from 'sentry/utils/useProjects';
+import {Referrer} from 'sentry/views/insights/agentMonitoring/utils/referrers';
+import {useEAPSpans} from 'sentry/views/insights/common/queries/useDiscover';
+
+function useOnboardingProject() {
+  const {projects} = useProjects();
+  const pageFilters = usePageFilters();
+  const selectedProject = getSelectedProjectList(
+    pageFilters.selection.projects,
+    projects
+  );
+  const agentMonitoringProjects = selectedProject.filter(p =>
+    agentMonitoringPlatforms.has(p.platform as PlatformKey)
+  );
+
+  if (agentMonitoringProjects.length > 0) {
+    return agentMonitoringProjects[0];
+  }
+  return selectedProject[0];
+}
+
+function useAgentEventWaiter(project: Project) {
+  const [refetchKey, setRefetchKey] = useState(0);
+
+  const {data, isPending} = useEAPSpans(
+    {
+      search: 'span.op:"gen_ai.*"',
+      fields: ['id'],
+      limit: 1,
+      enabled: !!project,
+      useQueryOptions: {
+        additonalQueryKey: [`refetch-${refetchKey}`],
+      },
+    },
+    Referrer.ONBOARDING
+  );
+
+  const hasEvents = Boolean(data?.length);
+
+  // Create a custom key that changes every 5 seconds to trigger refetch
+  // TODO(aknaus): remove this and add refetchInterval to useEAPSpans
+  useEffect(() => {
+    if (!hasEvents) {
+      const interval = setInterval(() => {
+        setRefetchKey(prev => prev + 1);
+      }, 5000); // Poll every 5 seconds
+
+      return () => clearInterval(interval);
+    }
+    return () => {};
```

Comment:
The control flow becomes easier to follow if you discard the disabled state first
```suggestion

    // Return early
    if (hasEvents) return () => {}
    
    // This is now the success case
    const interval = setInterval(() => {
      setRefetchKey(prev => prev + 1);
    }, 5000);

    return () => clearInterval(interval);
```